contrib_bots/lib/sched.py

```diff
@@ -0,0 +1,209 @@
+
+# Verify is hour format is valid
+def is_time_format(format):
+    try:
+        time.strptime(format, '%H:%M')
+        return True
+    except ValueError:
+        return False
+
+# For every task we want to schedule we define a job that have a random unique id  to cancel the task.
+
+def job(randomint, message, client):
+    original_content = message['content']
+    reduced_message_list = original_content.split()[3:]
+    reduced_message_string = " ".join(reduced_message_list)
+    if reduced_message_string:
+        if reduced_message_string.split()[0] == "at":
+            reduced_message_string = reduced_message_string.split()[2:]
+            reduced_message_string = " ".join(reduced_message_string)
+
+    events = dict(
+        type='stream',
+        to='alert',
+        subject="All alerts",
+        content=reduced_message_string+" "+randomint,
+        )
+    client.send_message(events)
+
+def start_up(client):
+    events = dict(
+        type='stream',
+        to='alert',
+        subject="All alerts",
+        content="@recover 0",
+        )
+    client.send_message(events)
+    return schedule.CancelJob
+
+class SchedHandler(object):
+    '''
+    This bot allows you to schedule daily alerts. It
+    looks for messages starting with '@every' or '@cancel' or '@recover'.
+
+    In this example, we write alerts at certain time intervals
+    in a new stream called "alert,".
+    '''
+
+    def usage(self, client, state_handler):
+        def run_threaded(job_func):
+            job_thread = threading.Thread(target=job_func)
+            job_thread.start()
+
+        def event_handler():
+            time.sleep(wakeup["timeout_in_seconds"]+4)
+            schedule.run_pending()
+        schedule.every(wakeup["timeout_in_seconds"]).seconds.do(run_threaded, event_handler)
+        time.sleep(wakeup["timeout_in_seconds"]+1)
+        schedule.run_pending()
+        time.sleep(wakeup["timeout_in_seconds"]+7)
+        schedule.every(1).seconds.do(start_up, client)
+
+        return '''
+            Users should preface messages with "@every" or '@cancel' or '@recover'.
+
+            Before running this, make sure to create a stream
+            called "alert" that your API user can send to.
+            '''
+
+    def triage_message(self, message):
+        # return True if we want to (possibly) response to this message
+
+        original_content = message['content']
+
+        # This next line of code is defensive, as we
+        # never want to get into an infinite loop of posting follow
+        # ups for own follow ups.
+        is_link = (original_content.startswith('@every') or
+                   original_content.startswith('@cancel') or
+                   original_content.startswith('@recover'))
+
+        return is_link
+
+    def handle_message(self, message, client, state_handler):
+
+        original_content = message['content']
+        original_sender = message["sender_email"]
+        a = str(randint(1, 1000000))
```

Comment:
So after a maximum of 1 million events you are going to run into problems. How about using UUIDs:
https://en.wikipedia.org/wiki/Universally_unique_identifier
https://docs.python.org/3.1/library/uuid.html